File: iqchannels-sdk/src/main/java/ru/iqchannels/sdk/ui/ButtonsAdapter.java
package ru.iqchannels.sdk.ui;

import android.os.Build;
import android.util.TypedValue;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.LinearLayout;

import androidx.annotation.NonNull;
import androidx.core.content.ContextCompat;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

import ru.iqchannels.sdk.R;
import ru.iqchannels.sdk.schema.SingleChoice;

public class ButtonsAdapter extends RecyclerView.Adapter<ButtonsAdapter.ButtonsVH> {

    private List<SingleChoice> items;

    private final ClickListener clickListener;

    ButtonsAdapter(ClickListener clickListener) {
        this.clickListener = clickListener;
    }

    @NonNull
    @Override
    public ButtonsVH onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        Button btn = new Button(parent.getContext());
        ViewGroup.MarginLayoutParams lp = new ViewGroup.MarginLayoutParams(
            LinearLayout.LayoutParams.MATCH_PARENT,
            LinearLayout.LayoutParams.WRAP_CONTENT
        );
        lp.setMargins(0, UiUtils.toPx(4), 0, 0);
        btn.setLayoutParams(lp);
        btn.setBackgroundResource(R.drawable.bg_single_choice_btn);
        btn.setTextColor(ContextCompat.getColor(parent.getContext(), R.color.light_text_color));
        btn.setTextSize(TypedValue.COMPLEX_UNIT_SP, 16);
        btn.setAllCaps(false);
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            btn.setElevation(0f);
        }
        return new ButtonsVH(btn);
    }

    @Override
    public void onBindViewHolder(@NonNull ButtonsVH holder, int position) {
        SingleChoice item = items.get(position);
        holder.bind(item);
        holder.itemView.setOnClickListener(v -> {
            clickListener.onClick(item);
        });
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    public void setItems(List<SingleChoice> items) {
        this.items = items;
        this.notifyDataSetChanged();
    }

    static class ButtonsVH extends RecyclerView.ViewHolder {

        ButtonsVH(View itemView) {
            super(itemView);
        }

        public void bind(SingleChoice item) {
            Button btn = (Button) itemView;
            btn.setText(item.title);
        }
    }

    interface ClickListener {

        void onClick(SingleChoice item);
    }
}

File: iqchannels-sdk/src/main/java/ru/iqchannels/sdk/ui/ButtonsAdapter.kt
package ru.iqchannels.sdk.ui

import android.util.TypedValue
import android.view.View
import android.view.ViewGroup
import android.view.ViewGroup.MarginLayoutParams
import android.widget.Button
import android.widget.LinearLayout
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView
import ru.iqchannels.sdk.R
import ru.iqchannels.sdk.applyIQStyles
import ru.iqchannels.sdk.schema.SingleChoice
import ru.iqchannels.sdk.setBackgroundStyle
import ru.iqchannels.sdk.styling.IQStyles

class ButtonsAdapter internal constructor(
	private val clickListener: ClickListener
) : RecyclerView.Adapter<ButtonsAdapter.ButtonsVH>() {

	private var items: List<SingleChoice>? = null

	override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ButtonsVH {
		val btn = Button(parent.context)
		val lp = MarginLayoutParams(
			LinearLayout.LayoutParams.MATCH_PARENT,
			LinearLayout.LayoutParams.WRAP_CONTENT
		)
		lp.setMargins(0, UiUtils.toPx(5), 0, 0)
		btn.run {
			layoutParams = lp
			minHeight = 0
			minimumHeight = 0
			setPadding(0, UiUtils.toPx(8), 0, UiUtils.toPx(8))
			setBackgroundStyle(
				IQStyles.iqChannelsStyles?.singleChoiceBtnStyles?.backgroundButton,
				IQStyles.iqChannelsStyles?.singleChoiceBtnStyles?.borderButton,
				R.color.color_single_choice_btn,
				R.color.color_single_choice_btn_border,
				1,
				4f
			)
			setTextColor(ContextCompat.getColor(parent.context, R.color.light_text_color))
			setTextSize(TypedValue.COMPLEX_UNIT_SP, 14f)
			applyIQStyles(IQStyles.iqChannelsStyles?.singleChoiceBtnStyles?.textButton)
			isAllCaps = false
			elevation = 0f
			stateListAnimator = null
		}

		return ButtonsVH(btn)
	}

	override fun onBindViewHolder(holder: ButtonsVH, position: Int) {
		val item = items?.get(position) ?: return
		holder.bind(item)
		holder.itemView.setOnClickListener { clickListener.onClick(item) }
	}

	override fun getItemCount(): Int {
		return items?.size ?: 0
	}

	fun setItems(items: List<SingleChoice>?) {
		this.items = items
		notifyDataSetChanged()
	}

	class ButtonsVH(itemView: View?) : RecyclerView.ViewHolder(itemView!!) {
		fun bind(item: SingleChoice) {
			val btn = itemView as Button
			btn.text = item.title
		}
	}

	internal interface ClickListener {
		fun onClick(item: SingleChoice)
	}
}
